backend/src/main/java/com/google/sps/servlets/AddParticipantServlet.java

```diff
@@ -0,0 +1,125 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.Match;
+import com.google.sps.data.Participant;
+import com.google.sps.FindMatchQuery;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+* Servlet that returns some example content.
+*/
+@WebServlet("/add-participant")
+public class AddParticipantServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Request parameter values
+    // TODO: Error check input values (before allowing to submit)
+    UserService userService = UserServiceFactory.getUserService();
+    String email = userService.getCurrentUser().getEmail();
+    String ldap = email.split("@")[0];
+    long timeAvailableUntil = Long.parseLong(request.getParameter("timeAvailableUntil"));
+    String timezone = request.getParameter("timezone");
+    int duration = Integer.parseInt(request.getParameter("duration"));
+    long timestamp = System.currentTimeMillis();
+    
+    // Set properties of entity
+    Entity participantEntity = new Entity("Participant");
+    participantEntity.setProperty("ldap", ldap);
+    participantEntity.setProperty("timeAvailableUntil", timeAvailableUntil);
+    participantEntity.setProperty("timezone", timezone);
+    participantEntity.setProperty("duration", duration);
+    participantEntity.setProperty("timestamp", timestamp);
+
+    // Insert entity into datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(participantEntity);
+
+    // Find match if possible or add to datastore.
+    FindMatchQuery findMatchQuery = new FindMatchQuery();
+    Match match = findMatchQuery.findMatchQuery(getParticipants());
+    if (match != null) {
+      addMatchToDatastore(match, datastore);
+      deleteParticipantsFromList(match, datastore);
+    }
```

Comment:
What do you think about moving this part (and the corresponding methods it calls) to the `SearchMatchServlet` to separate out the functionalities of the post and finding/searching for a match? I feel like this can help in modularization and may also be able to eliminate the "brute force search for a match" on line 68 of the `SearchMatchServlet` since you will be doing this part there instead. On the front-end, I will make sure to both post to the `AddParticipantServlet` and also fetch information about a match from the `SearchMatchServlet` every time the form is submitted.

Reply:
The reason why I have finding a match in the AddParticipantServlet part is because when you add a new participant, our algorithm calls for immediately checking if that new participant has a match. If not, then they're added to the datastore. If they do, then they're never added to the datastore of participants and just added as a match to the match datastore that's then searched in SearchMatchServlet. I don't think it would make sense to have finding and searching together.